### PR TITLE
[SofaNG] Setup GUI

### PR DESCRIPTION
**Pull request by fredroy**
#1527 

based on:
- #2843  

Deprecates the rest of **SofaGraphComponent** and **SofaUserInteraction**

Create a GUI (out of Sofa.Component) module, with:
 - **Common** (definition of GUIs)
 - GUIs themselves: **Batch**, **Qt** and **Headlessrecorder**
 - **Component**, with components only used for GUI (settings linked to performers)

Metamodule SofaGui is still used for runSofa (so with a kind of conditional factory, depending of the compilation option at cmake) but I think we should move this "conditional" compilation directly in runSofa

[~~ci-depends-on https://github.com/sofa-framework/SofaGLFW/pull/31~~]

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by guparan** [comment]
[ci-build][force-full-build][with-all-tests]

File: GUI/Qt/Sofa.GUI.QtConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+set(SOFAGUIQT_HAVE_QGLVIEWER @SOFAGUIQT_HAVE_QGLVIEWER@)
+set(SOFAGUIQT_HAVE_QT5_CHARTS @SOFAGUIQT_HAVE_QT5_CHARTS@)
+set(SOFAGUIQT_HAVE_QT5_WEBENGINE @SOFAGUIQT_HAVE_QT5_WEBENGINE@)
+set(SOFAGUIQT_HAVE_NODEEDITOR @SOFAGUIQT_HAVE_NODEEDITOR@)
+set(SOFAGUIQT_HAVE_TINYXML @SOFAGUIQT_HAVE_TINYXML@)
+
+# This is needed to be able to find plugins/SofaGuiQt/lib/cmake/QGLViewer
+list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_LIST_DIR}/..")
+
+
```
Comment (guparan) [inline GUI/Qt/Sofa.GUI.QtConfig.cmake.in]:
```suggestion
find_package(Sofa.GUI.Common QUIET REQUIRED)
```

File: GUI/Batch/Sofa.GUI.BatchConfig.cmake.in [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+find_package(SofaBase QUIET REQUIRED)
+find_package(SofaUserInteraction QUIET REQUIRED)
+find_package(SofaGraphComponent QUIET REQUIRED)
+find_package(SofaSimulationCommon QUIET REQUIRED)
+sofa_find_package(cxxopts QUIET REQUIRED)
+
+if(SOFAGUICOMMON_HAVE_SOFA_GL)
+    find_package(Sofa.GL QUIET REQUIRED)
+endif()
```
Comment (guparan) [inline GUI/Batch/Sofa.GUI.BatchConfig.cmake.in]:
```suggestion
find_package(Sofa.GUI.Common QUIET REQUIRED)
```

**Review by guparan — CHANGES_REQUESTED** [review]

**Pull request merged**